copy/single.go

```diff
@@ -131,12 +137,12 @@ func (c *copier) copySingleImage(ctx context.Context, unparsedImage *image.Unpar
 		src:             src,
 		// diffIDsAreNeeded is computed later
 		cannotModifyManifestReason:    cannotModifyManifestReason,
-		requireCompressionFormatMatch: requireCompressionFormatMatch,
+		requireCompressionFormatMatch: opts.requireCompressionFormatMatch,
 	}
 	if c.options.DestinationCtx != nil {
 		// Note that compressionFormat and compressionLevel can be nil.
-		ic.compressionFormat = c.options.DestinationCtx.CompressionFormat
-		ic.compressionLevel = c.options.DestinationCtx.CompressionLevel
+		ic.compressionFormat = opts.compressionFormat
+		ic.compressionLevel = opts.compressionLevel
 	}
```

Comment:
For our goals, it doesn‘t work to gate `ic.compression…` use on `c.options.DestinationCtx != nil` (assuming we won’t create a clone of `DestinationCtx`; and if we did create that clone, we wouldn’t need the new fields in `opts` anyway) — the clone will just set values in `opts`, and `DestinationCtx` could well be `nil`.

---

Also the way the responsibility for `options.DestinationCtx` is moved to callers now, the logic needs to be in two places.

I think this can be, in `copySingleImage`:
```go
if opts.compressionFormat != nil {
  ic.compression$both = opts.compression$both
} else if `c.options.DestinationCtx != nil {
  // existing comment
  ic.compression$both = c.options.DestinationCtx.Compression$both
}
```

Reply:
Nice catch! I have fixed it now.